### PR TITLE
Add --public/--private/--both flags when removing keys

### PR DESCRIPTION
**Pull request by JasonYangShadow**
## Description of the Pull Request (PR):
As requested in this issue
https://github.com/apptainer/apptainer/issues/465
to support removing private keys from apptainer keyring.

### This fixes or addresses the following GitHub issues:

 - Fixes https://github.com/apptainer/apptainer/issues/465

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/apptainer/apptainer/blob/main/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)
- Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see CONTRIBUTING.md).
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTORS.md)

### CODE REVIEW

**Comment by JasonYangShadow** [comment]
@cclerget @ikaneshiro @DrDaveD 
Sorry to bother you, but please help review this PR if you have time. 
Feel free to leave comments and I will resolve them ASAP. 
🙇🏽 😄

File: pkg/sypgp/sypgp.go [hunk 1]
```diff
@@ -494,6 +516,27 @@ func (keyring *Handle) RemovePubKey(toDelete string) error {
...
+	case err != nil:
+		return fmt.Errorf("unable to list local secret keyring: %v", err)
+	}
+
+	newKeyList := removeKey(elist, toDelete)
+	if newKeyList == nil {
+		return fmt.Errorf("no key matching given fingerprint found")
+	}
+
+	sylog.Verbosef("updating local secret keyring: %v", keyring.SecretPath())
```
Comment (cclerget) [inline pkg/sypgp/sypgp.go:535]: Just a missing uppercase here

Comment (JasonYangShadow) [reply pkg/sypgp/sypgp.go]: @cclerget thanks for the comment. Resolved the comment in this commit
[`ab65928` (#555)](https://github.com/apptainer/apptainer/pull/555/commits/ab65928bb0818a107ab8f68af1245bca498ef2ef)

**Review by cclerget — APPROVED** [review]
LGTM

**Pull request merged**